compiler/tests-spec/testData/diagnostics/notLinked/contracts/declarations/contractFunction/neg/2.fir.kt

```diff
@@ -43,7 +43,7 @@ class case_4 : ClassLevel3() {
 
     fun <T>T.case_4_3_wrap() {
         fun case_4_3_contract() {
-            contract { returns() implies (this@case_4_3_wrap is ClassLevel1) }
+            contract { returns() implies (<!USELESS_IS_CHECK!>this@case_4_3_wrap is ClassLevel1<!>) }
```

Comment:
This seems false warning, but smartcast for `this` is `ClassLevel1 & T`. Didn't dig further why smartcast has that, but I suspect that such info is back-propagated from the below `this@... !is ClassLevel1` (if persistent flow is not isolated)?